api/openai.py

```diff
@@ -46,95 +45,89 @@ def __init__(self):
         self.log = get_logger()
 
     def is_channel_allowed(self, message: ServiceMessage) -> bool:
-        if message.service in openai_channel_ids and message.channel.id in openai_channel_ids[message.service]:
-            # For Rob's discord
-            return True
-        elif paid_service_all_channels:
-            return True
-        elif message.channel.id in paid_service_channel_ids:
-            # if list is empty, default
-            return True
+        channel_id = (message.channel and message.channel.id)
+        return (
+            paid_service_all_channels or
+            channel_id in openai_allowed_sources.get(message.service.value, [])
+        )
+
+    def log_error(self, error, exception=None, warning=False):
+        if warning:
+            self.log.warning(self.class_name, error=error)
         else:
-            return False
-    def is_text_risky(self, text: str) -> bool:
-        """Ask the openai moderation endpoint if the text is risky
+            self.log.error(self.class_name, error=error)
 
-        See https://platform.openai.com/docs/guides/moderation/quickstart for details"""
+        loop = asyncio.get_running_loop()
+        loop.create_task(utils.log_error(error))
+        if exception:
+            loop.create_task(utils.log_exception(exception))
+
+    def is_text_risky(self, text: str) -> bool:
+        """Ask the openai moderation endpoint if the text is risky.
 
-        allowed_categories = frozenset("violence") # Can be triggered by some AI safety terms
+        See https://platform.openai.com/docs/guides/moderation/quickstart for details.
+        """
+        allowed_categories = {"violence"} # Can be triggered by some AI safety terms
```

Comment:
@ProducerMatt - could you please include a list of the known AI safety terms that trigger the violence thingy? I don't like this blank check to allow violence because of "some" terms...